service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -217,9 +217,12 @@ private String buildNotePertinentInformation(Immunization immunization) {
     private String buildExplanationPertinentInformation(Immunization immunization) {
         Optional<String> explanation;
         if (immunization.hasExplanation() && immunization.getExplanation().hasReason()) {
-            CodeableConcept reason = immunization.getExplanation().getReasonFirstRep();
-            explanation = CodeableConceptMappingUtils.extractTextOrCoding(reason);
-            return explanation.map(value -> REASON + value).orElse(StringUtils.EMPTY);
+            String reasonGiven = REASON + immunization.getExplanation().getReason().stream()
+                .map(CodeableConceptMappingUtils::extractTextOrCoding)
+                .map(Optional::get)
+                .filter(value -> !value.equals(StringUtils.EMPTY))
+                .collect(Collectors.joining(StringUtils.SPACE));
+            return reasonGiven.equals(REASON) ? StringUtils.EMPTY : reasonGiven;
```

Comment:
This is a bit unsafe. I would build a string with reasons and add the REASON only if the list is not empty.
```
return StringUtils.EMPTY.equals(reasons) ? StringUtils.EMPTY : (REASON + reasons);
```

Reply:
Addressed